polyglot/piranha/src/piranha.rs

```diff
@@ -59,76 +72,146 @@ impl SourceCodeUnit {
     }
   }
 
-  // Applies the rule to the first match in the source code
-  // This is implements the main algorithm of piranha.
+  /// Applies the rule to the first match in the source code
+  /// This is implements the main algorithm of piranha.
+  /// Parameters:
+  /// * `rule` : the rule to be applied
+  /// * `rule_store`: contains the input rule graph.
+  ///
+  /// Algo:
```

Comment:
Unless this is a Rust convention I am unaware of, let's just use "Algorithm" in full.

Reply:
Lol! its not. Thanks for pointing out. I ll make a change.